javascripts/main.js

```diff
@@ -75,3 +77,15 @@ input.addEventListener('input', inputChange);
 form.addEventListener('submit', post);
 tweetList.addEventListener('click', remove);
 tweetList.addEventListener('click', like);
+
+(function() {
+  dbRefObject.on('value', function(snap) {
+    var object = snap.val();
+    for (var item in object) {
+      var tweet = object[item];
+      tweets.push(tweet);
```

Comment:
This is your problem... every time that you add an item you are looping over all the items in the array and pushing them into `tweets`. `push` just always adds to the end of the object/array so you want to update this to be something like so:

```
(function() {
  dbRefObject.on('value', function(snap) {
    let object = snap.val();
    let newTweetArr = [];
    for (let item in object) {
      let tweet = object[item];
      newTweetArr.push(tweet);
    }
    tweets = newTweetArr;
    updateTweets();
  });
}());
```
(You will need to change `const tweets` to be `let tweets`)

Reply:
You legend. That's what I was after!